austrakka/components/sequence/funcs.py

```diff
@@ -523,12 +524,36 @@ def _get_files_from_csv_paths(csv_row: pd.Series, seq_type: SeqType) -> List[Seq
         if column==SEQ_ID_CSV:
             continue
         contig_rename_id = seq_id if seq_type==SeqType.FASTA_ASM else None
-        file = _get_file(csv_row[column], contig_rename_id)
+        read_hint = _col_name_to_read_hint(column)
+        file = _get_file(csv_row[column], contig_rename_id, read_hint)
+        _ensure_read_hint_and_filename_agrees(file)
         sample_files.append(file)
     return sample_files
 
 
-def _get_file(filepath: str, contig_rename_id=None) -> SeqFile:
+def _col_name_to_read_hint(column):
+    read_hint = None
+    if column == PATH_1_CSV:
+        read_hint = 'r1'
+    elif column == PATH_2_CSV:
+        read_hint = 'r2'
+    return read_hint
+
+
+def _ensure_read_hint_and_filename_agrees(file):
+    file_path = Path(file.filename)
+    file_name = file_path.stem.casefold()
+    
+    if file.read_hint == "r1" and file_name.endswith("_r2"):
```

Comment:
Checked by the server so this check isn't critical, but really ought to do a case-insensitive check here (e.g. file_name.lower().endswith()) since filenames very often end with _R1 or _R2